File: backend/src/main/java/com/backend/ord/utils/Console.java
package com.backend.ord.utils;

import com.backend.ord.enums.ConsoleColor;

import java.util.function.Function;
import java.util.stream.IntStream;

public final class Console {
    private Console() {
    }

    public static void print(String message) {
        System.out.print(message);
    }

    public static void print(String message, ConsoleColor color) {
        String msg = String.format("%s%s%s", color.getANSICode(), message, ConsoleColor.Reset.getANSICode());
        System.out.print(msg);
    }

    public static void printBlue(String message) {
        print(message, ConsoleColor.Blue);
    }

    public static void printRed(String message) {
        print(message, ConsoleColor.Red);
    }

    public static void printGreen(String message) {
        print(message, ConsoleColor.Green);
    }

    public static void printYellow(String message) {
        print(message, ConsoleColor.Yellow);
    }

    public static void printPurple(String message) {
        print(message, ConsoleColor.Purple);
    }

    public static void printCyan(String message) {
        print(message, ConsoleColor.Cyan);
    }

    public static void addBreakLine(int lines) {
        IntStream.range(0, lines).forEach(i -> System.out.println());
    }

    public static void ensureFunctionSuccess(String introMsg, Runnable function) {
        print(introMsg);
        try {
            function.run();
            Console.printGreen(" ✅ DONE\n");
        } catch (Exception e) {
            Console.printRed(" ❌ ERROR\n");
        }
    }
}

File: backend/src/main/kotlin/com/backend/ord/utils/Console.kt
package com.backend.ord.utils



object Console {
    enum class Color(val ansiCode: String) {
        Blue("\u001B[34m"),
        Red("\u001B[31m"),
        Green("\u001B[32m"),
        Yellow("\u001B[33m"),
        Purple("\u001B[35m"),
        Cyan("\u001B[36m"),
        Reset("\u001B[0m")
    }

    fun print(message: String?) {
        kotlin.io.print(message)
    }

    fun print(message: String?, color: Color) {
        val msg = "${color.ansiCode}$message${Color.Reset.ansiCode}"
        kotlin.io.print(msg)
    }

    fun printBlue(message: String?) = print(message, Color.Blue)
    fun printRed(message: String?) = print(message, Color.Red)
    fun printGreen(message: String?) = print(message, Color.Green)
    fun printYellow(message: String?) = print(message, Color.Yellow)
    fun printPurple(message: String?) = print(message, Color.Purple)
    fun printCyan(message: String?) = print(message, Color.Cyan)

    fun addBreakLine(lines: Int) {
        repeat(lines) { println() }
    }

    fun ensureFunctionSuccess(introMsg: String?, function: () -> Any) {
        print(introMsg)
        try {
            val outroMsg = function()
            printGreen(" ✅ DONE\n")

            if (outroMsg is String) {
                println(outroMsg)
            }
        } catch (e: Exception) {
            printRed(" ❌ ERROR\n")

            throw e
        }
    }

    fun printWithMargin(message: String?) {
        addBreakLine(2)
        print(message)
        addBreakLine(2)
    }
}
